django/core/handlers/asgi.py

```diff
@@ -0,0 +1,348 @@
+        return self._post
+
+    def _set_post(self, post):
+        self._post = post
+
+    def _get_files(self):
+        if not hasattr(self, "_files"):
+            self._read_started = False
+            self._load_post_and_files()
+        return self._files
+
+    POST = property(_get_post, _set_post)
+    FILES = property(_get_files)
+
+    @cached_property
+    def COOKIES(self):
+        return http.parse_cookie(self.META.get("HTTP_COOKIE", ""))
+
+
+class AsgiHandler(base.BaseHandler):
+    """
+    Handler for ASGI requests.
+    """
+
+    request_class = AsgiRequest
+
+    # Size to chunk response bodies into for multiple response messages
+    chunk_size = 512 * 1024
+
+    def __init__(self):
+        super(AsgiHandler, self).__init__()
+        self.load_middleware()
+
+    async def __call__(self, scope, receive, send):
+        """
+        Async entrypoint - parses the request and hands off to get_response.
+        """
+        # Only serve HTTP connections (we should allow some way to override this)
+        if scope["type"] != "http":
+            raise ValueError(
+                "Django can only handle ASGI/HTTP connections, not %s"
+                % scope["type"]
+            )
+        # Receive the HTTP request body
+        try:
+            body = await self.read_body(receive)
+        except RequestAborted:
+            return
+        # If we've got here, then the request is complete and we can serve it.
+        set_script_prefix(self.get_script_prefix(scope))
+        await sync_to_async(signals.request_started.send)(sender=self.__class__, scope=scope)
+        # Get the request and check for basic issues
+        request, error_response = self.create_request(scope, body)
+        if request is None:
+            await self.send_response(error_response, send)
+            return
+        # Get the response, using a threadpool via sync_to_async if needed
+        if asyncio.iscoroutinefunction(self.get_response):
+            response = await self.get_response(request)
+        else:
+            # If get_response is synchronous, run it non-blocking
+            response = await sync_to_async(self.get_response)(request)
+        response._handler_class = self.__class__
+        # Increase chunk size on file responses (ASGI servers will handle low-level chunking)
+        if isinstance(response, FileResponse):
+            response.block_size = self.chunk_size
+        # Send the response
+        await self.send_response(response, send)
+
+    async def read_body(self, receive):
+        """
+        Reads a HTTP body from an ASGI connection.
+        """
+        body = b""
+        while True:
+            message = await receive()
+            if message["type"] == "http.disconnect":
+                # Bye bye.
+                raise RequestAborted()
+            else:
+                # See if the message has body, and if it's the end, launch into
+                # handling (and a synchronous subthread)
```

Comment:
I recon i'm a bit hesitating on this comment, does the following if-block concern "if it's the end" ? That's what i thought, but further down there is another " # Quit out if that's the end" comment so perhaps that one is more for the else-block than for the if-block.
But the next comment "Limit the maximum request data size" is above the if-block that it actually comments. 
Thanks for your support.

Reply:
You were very right - that comment leaked over from some old Channels code. I've revised it.